tests/saw/spec/handshake/handshake_io_lowlevel.saw

```diff
@@ -158,18 +208,54 @@ let setup_connection = do {
    crucible_points_to (ocsp_status_size cak) (crucible_term status_size);
    crucible_equal (crucible_term status_size) (crucible_term {{zero : [32]}});
 
-   config_bitfield_value <- crucible_fresh_var "config_bitfield" (llvm_int 16);
-   crucible_points_to (config_bitfield config) (crucible_term config_bitfield_value);
-   // The following line could be made more precise, we only care that the use_tickets
-   // bit is zero 
-   crucible_equal (crucible_term config_bitfield_value) (crucible_term {{zero : [16]}});
+   // Here, we initialize two bitfields: one in s2n_connection (conn_bitfield)
+   // and the other in s2n_config (config_bitfield). We start by initializing
+   // the entirety of each bitfield with a symbolic value of the appropriate
+   // size...
+   conn_bitfield_value <- crucible_fresh_var "conn_bitfield" (llvm_int conn_bitfield_size);
+   crucible_points_to_untyped (conn_bitfield pconn) (crucible_term conn_bitfield_value);
+   config_bitfield_value <- crucible_fresh_var "config_bitfield" (llvm_int config_bitfield_size);
+   crucible_points_to_untyped (config_bitfield config) (crucible_term config_bitfield_value);
+
+   // ...next, we impose preconditions on these symbolic values as appropriate.
+   //
+   // We assume that corking/uncorking is managed by s2n, so set the corked_io
+   // bit in s2n_connection to one...
+   let corked_io = {{ bitfield_is_set conn_bitfield_value conn_corked_io_index : Bit }};
+   crucible_precond corked_io;
+
+   // ...we assume that the client_session_resumed bit in s2n_connection must
+   // be zero...
+   crucible_precond {{ ~(bitfield_is_set conn_bitfield_value conn_client_session_resumed_index) }};
```

Comment:
If you're wondering why this PR imposes a precondition on `conn->client_session_resumed` when the specification didn't involve it before... actually, the specification _did_ involve it before! But it did so implicitly. This code was responsible:

https://github.com/aws/s2n-tls/blob/d5a044b8ce4eee8273397ee99d224faf8b562a39/tests/saw/spec/handshake/handshake_io_lowlevel.saw#L128-L130

`conn_corked_io pconn` points to a bitfield, and due the way SAW currently interacts with bitfields, it will initialize the first 8 bits of the bitfield to `1 : [8]`. This means that `corked_io`, the first bit, will be set to one, but the subsequent seven bits will be set to zero. As it turns out, `client_session_resumed` is one of these seven bits.

I discovered this quirk in the process of refactoring this bitfield-related code, so I decided to see what would happen if I didn't impose preconditions on any of these seven bits. If I do so, the proofs _almost_ work, but the proof of `s2n_conn_set_handshake_type` will fail. These lines explain why:

https://github.com/aws/s2n-tls/blob/d5a044b8ce4eee8273397ee99d224faf8b562a39/tls/s2n_handshake_io.c#L853-L873

Note that if `conn->client_session_resumed` is enabled, then it is possible for `s2n_conn_set_handshake_type` to return early without enabling `FULL_HANDSHAKE`, `TLS12_PERFECT_FORWARD_SECRECY`, or `OCSP_STATUS`.

Since the proof was already implicitly assuming that `conn->client_session_resumed` should be set to zero, I decided to preserve this convention in this PR. Alternatively, we could make it so that we do not impose any preconditions on `conn->client_session_resumed` at all. However, this would have some downstream effects on the SAW proofs. The Cryptol definition of `conn_set_pre_tls13_handshake_type` would need to change, and this would likely impact the `rfc_handshake_tls12` proofs as well.

Reply:
I think this theoretically isn't a huge new issue, since we're also apparently setting config->use_tickets to false. But if we're completely skipping proving session resumption logic, that's a pretty big miss in the proofs. Do we have an issue to remove the restrictions on config->use_tickets (and conn->client_session_resumed)?